internal/provider/resource_api_key.go

```diff
@@ -21,6 +21,7 @@ func resourceAPIKey() *schema.Resource {
 			"key_name": {
 				Type:        schema.TypeString,
 				Required:    true,
+				ForceNew:    true,
```

Comment:
**(info)** If no update is defined, the input's change must force recreate the resource.